### PR TITLE
[IMP] mail: place cursor at end of composer input on message edit

### PR DESCRIPTION
**Pull request by oomsveta**

### CODE REVIEW

File: addons/mail/static/src/new/core/composer_model.js [hunk 1]
```diff
@@ -55,6 +55,10 @@ export class Composer {
     update(data) {
         if ("textInputContent" in data) {
             this.textInputContent = data.textInputContent;
+            // place cursor at end of the content
```
Comment (seb-odoo) [inline addons/mail/static/src/new/core/composer_model.js]: The comment seems a bit overkill, also this part of the code doesn't really place the cursor at the end as its just saving a state in model.

Also I don't really understand why this is in model generic update rather in the method actually triggering an edit.

**Comment by oomsveta** [comment]
@seb-odoo changes done

File: addons/mail/static/src/new/core/composer_model.js [hunk 2]
```diff
@@ -56,6 +56,9 @@ export class Composer {
         if ("textInputContent" in data) {
             this.textInputContent = data.textInputContent;
         }
+        if ("selection" in data) {
+            this.selection = data.selection;
```
Comment (seb-odoo) [inline addons/mail/static/src/new/core/composer_model.js]: `Object.assign`

File: addons/mail/static/tests/new/message/message_tests.js [hunk 3]
```diff
@@ -51,6 +51,31 @@ QUnit.test("Start edition on click edit", async (assert) => {
...
+        discuss: {
+            context: {
+                active_id: `mail.channel_${channelId}`,
+            },
+        },
+    });
+    await openDiscuss();
+    await click("i[aria-label='Edit']");
+    const composerTextarea = target.querySelector(".o-mail-composer-textarea");
+    assert.strictEqual(composerTextarea.value.length, composerTextarea.selectionStart);
```
Comment (seb-odoo) [inline addons/mail/static/tests/new/message/message_tests.js]: assert on start, end, direction

**Comment by oomsveta** [comment]
@seb-odoo changes done

**Pull request merged**